src/Discord.Net/WebSocket/Entities/Guilds/SocketGuild.cs

```diff
@@ -307,6 +308,7 @@ public VoiceState AddOrUpdateVoiceState(VoiceStateModel model, DataStore dataSto
                 _audioLock.Release();
             }
             await audioClient.ConnectAsync(url, CurrentUser.Id, voiceState.VoiceSessionId, token).ConfigureAwait(false);
```

Comment:
Should also have a try-catch around .ConnectAsync. If it exceptions, feed it to the promise.